src/sparsezoo/model/utils.py

```diff
@@ -176,17 +203,17 @@ def filter_files(
         raise ValueError("No files found - the list of files is empty!")
 
     if num_recipe_file_dicts >= 2:
-        recipe_names = [
-            file_dict["display_name"]
-            for file_dict in files_filtered
-            if file_dict["file_type"] == "recipe"
-        ]
-        raise ValueError(
-            f"Found multiple recipes: {recipe_names}, "
-            f"for the string argument {expected_recipe_name}"
-        )
-    else:
-        return files_filtered
+        recipe_names = set()
```

Comment:
I guess you are trying to validate the ```files_filtered``` here by using the Set structure. If the whole checking is looking for duplicates, why not use a Dictionary to record all the ```file_dict``` that has been visited and immediately raise a ValueError when the current ```file_dict``` is found to be a duplicate?

Reply:
Good comment. In the new folder structure, the same recipe can be in different subfolders. In that case we are expected to have duplicates.

Reply:
I see... so what would happen if in this code we find a new value for ```file_dict["display_name"]```?

Reply:
Should it be like:
```
if file_dict["file_type"] == "recipe" and file_dict["display_name"] not in collected_display_names_dict:
    raise ValueError(......)
``` ?